src/vs/workbench/contrib/customEditor/browser/extensionContributedEditorService.ts

```diff
@@ -0,0 +1,343 @@
+import { EditorActivation, IEditorOptions, ITextEditorOptions } from 'vs/platform/editor/common/editor';
+import { registerSingleton } from 'vs/platform/instantiation/common/extensions';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+import { IStorageService, StorageScope, StorageTarget } from 'vs/platform/storage/common/storage';
+import { EditorAssociations, editorsAssociationsSettingId } from 'vs/workbench/browser/editor';
+import { EditorOptions, IEditorInput } from 'vs/workbench/common/editor';
+import { CustomEditorInfo } from 'vs/workbench/contrib/customEditor/common/customEditor';
+import { IEditorGroup, IEditorGroupsService } from 'vs/workbench/services/editor/common/editorGroupsService';
+import { IEditorService, IOpenEditorOverride, IOpenEditorOverrideEntry } from 'vs/workbench/services/editor/common/editorService';
+import { Schemas } from 'vs/base/common/network';
+import { ContributedEditorPriority, priorityToRank } from 'vs/workbench/contrib/customEditor/common/extensionContributedEditorService';
+import { DiffEditorInput } from 'vs/workbench/common/editor/diffEditorInput';
+
+enum ExtensionContributedEditorChoice {
+	DEFAULT = 1,
+	OPTIONAL = 2,
+}
+
+type ExtensionContributedEditorChoiceEntry = {
+	editorID: string;
+	choice: ExtensionContributedEditorChoice;
+};
+
+interface IContributedEditorInput extends IEditorInput {
+	viewType?: string;
+}
+
+interface ContributionPoint {
+	scheme: string | undefined,
+	globPattern: string,
+	priority: number,
+	editorInfo: ContributedEditorInfo,
+	options: ContributionPointOptions,
+	createEditorInput: (resource: URI, editorID: string, group: IEditorGroup) => IEditorInput,
+	createDiffEditorInput?: (diffEditorInput: DiffEditorInput, editorID: string, group: IEditorGroup) => IEditorInput
+}
+
+type ContributionPoints = Array<ContributionPoint>;
+
+export type ContributionPointOptions = {
+	singlePerGroup?: boolean | (() => boolean);
+	singlePerResource?: boolean | (() => boolean);
+};
+
+export type ContributedEditorInfo = {
+	id: string;
+	active: (currentEditor: IEditorInput) => boolean;
+	instanceOf: (editorInput: IEditorInput) => boolean;
+	label: string;
+	detail: string;
+	priority: ContributedEditorPriority;
+};
+
+export interface IExtensionContributedEditorService {
+	readonly _serviceBrand: undefined;
+	contributedEditorOverride(handler: IExtensionContributedEditorHandler): IDisposable;
+	getAssociationsForResource(resource: URI): EditorAssociations;
+	/**
+	 * Registers a specific editor contribution.
+	 * @param scheme The scheme of this contribution, if defined it takes precedent
+	 * @param globPattern The glob pattern for this contribution point
+	 * @param editorInfo Information about the contribution point
+	 * @param options Specific options which apply to this contribution
+	 * @param createEditorInput The factory method for creating inputs
+	 */
+	registerContributionPoint(
```

Comment:
This is a way too generic name for me. What is a contribution point when it comes to editors? Why not `registerEditorOverride`?